cdap-standalone/src/main/java/co/cask/cdap/StandaloneMain.java

```diff
@@ -386,6 +402,7 @@ public static void main(String[] args) {
         System.err.println("Failed to start Standalone CDAP");
         e.printStackTrace(System.err);
       }
+      main.stopExternalProcesses();
```

Comment:
to clarify the discussion is around the next line, `Runtime.getRuntime().halt(-2);` why we use -2 code there. not sure we have a clear answer for that.